app/uk/gov/hmrc/helptosavefrontend/controllers/BankAccountController.scala

```diff
@@ -68,17 +69,30 @@ class BankAccountController @Inject() (val helpToSaveService:     HelpToSaveServ
       withErrors ⇒
         Ok(views.html.register.bank_account_details(withErrors, selectEmailPage)),
       { bankDetails ⇒
-        checkIfAlreadyEnrolledAndDoneEligibilityChecks(htsContext.nino) {
-          session ⇒
-            sessionCacheConnector.put(session.copy(bankDetails = Some(bankDetails)))
-              .fold(
-                error ⇒ {
-                  logger.warn(s"Could not update session with bank details: $error")
-                  internalServerError()
-                },
-                _ ⇒ SeeOther(routes.RegisterController.checkDetails().url)
-              )
-        }
+        barsService.validate(bankDetails).map[Future[Result]] {
+          case Right(true) ⇒
+            checkIfAlreadyEnrolledAndDoneEligibilityChecks(htsContext.nino) {
+              session ⇒
+                val sortCodeWithHyphens = bankDetails.sortCode.grouped(2).mkString("-")
+                sessionCacheConnector.put(session.copy(bankDetails = Some(bankDetails.copy(sortCode = sortCodeWithHyphens))))
```

Comment:
is this not an overkill for sortcode ? because by this point we are sure that the input has 6 digits in it without any spaces so `.grouped(2).mkString("-")` looks easier ?

Reply:
the type doesn't guarantee that that's the case - the type is `String` which can be anything. It feels safer to not rely on the String to be a particular format as assumptions can be wrong